### PR TITLE
Add v7 upgrade guide

### LINKED ISSUE #200: Add v7 Upgrade guide
Add documentation to cover upgrading to v7. 

Acceptance Criteria:
- [ ]

### PR DESCRIPTION
**Pull request by roborourke**
Fixes #200

### CODE REVIEW

File: other-docs/guides/upgrading/v7.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+If you have any code referencing the class `S3_Uploads` this will need to be updated to `S3_Uploads\Plugin`. Additionally any other references to classes starting with `S3_Uploads` should be updated to use `S3_Uploads` as the namespace, for example `S3_Uploads_Local_Stream_Wrapper` becomes `S3_Uploads\Local_Stream_Wrapper`. Any functions prefixed with `s3_uploads_*` have also been moved into the `S3_Uploads` namespace, for example `s3_uploads_enabled()` is now `S3_Uploads\enabled()`.
+
+All constants remain unchanged.
+
+### Altis Experiments
+
+The experiments feature of native analytics has been merged into the main native analytics codebase. This means any references in your code to the namespace `Altis\Experiments` should be updated to `Altis\Analytics\Experiments`.
+
+Additionally the `Altis\Analytics\Experiments\register_goal()` function accepts a different set of arguments than it did in Altis v6.
```
Comment (joehoyle) [inline other-docs/guides/upgrading/v7.md:83]: Could you summarize or link to that change?

File: other-docs/guides/upgrading/v7.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+### Experience Block Insights
+
+Continuing from previous improvements to tracking the performance of Experience Blocks in v6, Altis v7 adds a dedicated dashboard for showing all Experience Blocks across the site with options for sorting by views and conversion rate.
+
+When editing blocks warnings are shown to authors if their selected conversion goals can not be tracked, and if they are missing fallback content.
+
+In addition there is a new, more detailed overview of Experience Block data that more clearly shows audience coverage, views and conversion rate changes over time and also detailed breakdowns of how each variant is performing compared to the fallback.
+
+On the software level several new REST API endpoints have been added for fetching blocks and block anlaytics data, along with a background post type that can be used for storing persistent data.
```
Comment (joehoyle) [inline other-docs/guides/upgrading/v7.md:107]:
```suggestion
Several new REST API endpoints have been added for fetching blocks and block anlaytics data, along with a background post type that can be used for storing persistent data.
```

File: other-docs/guides/upgrading/v7.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+### Authorship
+
+Altis now supports adding multiple authors to posts as well as managing guest authors, either as full or partial users.
+
+This feature was funded by and built in collaboration with Siemens.
```
Comment (joehoyle) [inline other-docs/guides/upgrading/v7.md:113]: Do we have a docs page for this?

File: other-docs/guides/upgrading/v7.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+### Local Server
+
+- Added a `mysql:` link to the output from `composer server db info` for software that supports the format. This allows using command or control click in the terminal to open the link.
+- A new `composer server db exec` command allows executing SQL from the command line without logging into the container or using some other tool
+- Experimental support for [Mutagen.io](https://mutagen.io) volume sharing. This can improve speed and file synchronisation issues on MacOS and Windows in some cases
+
+### Search
+
+- User and terms indexables can now be turned on or off via config
```
Comment (joehoyle) [inline other-docs/guides/upgrading/v7.md:135]: Could we link to updated docs for that?

File: other-docs/guides/upgrading/v7.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+- Better handling for large requests such as documents during indexing
+- Improved default search algorithm with a tighter scope for matches
+
+### Other Changes
+
+Altis v7 includes many other small changes, consult the changelog for full details of all the changes shipped in this release.
```
Comment (joehoyle) [inline other-docs/guides/upgrading/v7.md:141]: Can we link the full changeling?

File: other-docs/guides/upgrading/v7.md (hunk 3 above)
Comment (roborourke) [reply other-docs/guides/upgrading/v7.md:113]: yeah good shout, I'll link to that above but thinking about it now I might leave this sentence at the bottom out, there's still a few things being agreed there

File: other-docs/guides/upgrading/v7.md (hunk 5 above)
Comment (roborourke) [reply other-docs/guides/upgrading/v7.md:141]: It doesn't exist yet as we haven't done the release so it would have to be added later

File: other-docs/guides/upgrading/v7.md (hunk 4 above)
Comment (roborourke) [reply other-docs/guides/upgrading/v7.md:135]:
```suggestion
- [User and terms indexables can now be turned on or off via config](docs://search/#additional-configuration-options)
```

File: other-docs/guides/upgrading/v7.md (hunk 3 above)
Comment (roborourke) [reply other-docs/guides/upgrading/v7.md:113]:
```suggestion
[Full documentation for the Authorship feature including how to enable it can be found here](docs://cms/authorship/).
```

File: other-docs/guides/upgrading/v7.md (hunk 1 above)
Comment (roborourke) [reply other-docs/guides/upgrading/v7.md:83]:
```suggestion
Additionally the `Altis\Analytics\Experiments\register_goal()` function accepts a different set of arguments than it did in Altis v6, [see the Goal Tracking documentation for full details](docs://analytics/native/goal-tracking/#php).
```

**Comment by roborourke** [comment]
Thanks @joehoyle, added those suggestions and made a note to update the last section with a link to the release changelog after it's been cut.

**Review by joehoyle — APPROVED** [review]

**Comment by joehoyle** [comment]
Sounds good, thanks @roborourke
